packages/react/src/ActionList/Item.tsx

```diff
@@ -95,7 +98,7 @@ export const Item = React.forwardRef<HTMLLIElement, ActionListItemProps>(
 
     const onSelect = React.useCallback(
       (
-        event: React.MouseEvent<HTMLLIElement> | React.KeyboardEvent<HTMLLIElement>,
+        event: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>,
```

Comment:
I wonder if loosening the type here (which I think is a good way to eliminate the integration issue with dotcom) might have any "tangible" disadvantages? 🤔

Reply:
It's definitely not as strict, but it does ease the integration in dotcom. I think with the feature flag it makes it a bit trickier as the type is dependent on the FF being enabled. The issue I had when I was dealing with TS early on was that there are two potential things it could be; a list item or a button. In addition to that, the `<li>` will always be the parent to anything else within the `item`, so technically it is an `HTMLLIElement`, but also a `HTMLButtonElement` since everything is being applied to the button, not the `<li>`.

I did try adding both types `HTMLLIElement | HTMLButtonElement` [and I got odd issues in dotcom](https://github.com/primer/react/pull/4272#discussion_r1529075714) 🤔

Reply:
Yeah I remember that! Let's see how this type works then and since it is under ff, we can always work on it!

Reply:
Hello from the future! This came up during release conductor stuff this week in dotcom, it seems like this change will cause typed handlers to fail since they are using the more specific type.

Since we've switched to a more generic type, these more specific ones will fail 😭

Reply:
Hey Josh! This was a tricky situation since the current type `HTMLLIElement` is technically correct, but also isn't depending on the context. Adding both the LI type and Button type caused a different issue ☹️. 

I made the types more generic in dotcom in my integration test PR that should clean up the failures you're seeing. It does involve making the type a bit more generic, but we don't seem to lose out on too much. Let me know what you think though! 

* [Original integration test PR with more context](https://github.com/github/github/pull/317139#issuecomment-2004140416)
* [Most recent integration test PR with changes](https://github.com/github/github/pull/329335#issuecomment-2181252241)